rpc/src/custom_rpc_server.rs

```diff
@@ -80,15 +78,12 @@ pub fn start_rpc_server<S: Storage + Send + Sync + 'static>(
 
     let server = Server::bind(&rpc_addr).serve(service);
 
-    task::spawn(async move {
-        server.await.expect("The RPC server couldn't be started!");
+    std::thread::spawn(move || {
```

Comment:
The `hyper`-backed RPC server is non-blocking, and under the main `tokio` threadpool; what's the benefit of a dedicated thread here just to block on its task?

Reply:
Ok, I see that some of the RPCs use `block_on`; this means that instead, we should use `task::spawn_blocking` here instead of `thread::spawn`.

Reply:
we can then use `block_on` executors to get around the shitty RPC framework not supporting async calls

Reply:
Yeah, but we can should those within blocking tasks, no? And right now, even if we do launch an extra thread, we are still within the same executor and might have the `block_on` done within a non-blocking task from the `tokio` threadpool.

Reply:
As discussed in the call, the RPC calls are no longer blocking, correct? In this case this can remain a normal `task`.